css: Disable view type dropdown for spectators in channel topics list

Hell1213 opened the pull request:
Spectators (logged out users) can't use the view-type dropdown in the 'list of topics' view for channels. This adds the missing CSS styling to make the dropdown appear disabled (grayed out with not-allowed cursor) for spectators, matching the behavior in recent conversations view.

**Fixes:** #36496

**How changes were tested:**
- Ran CSS linting: `./tools/lint --only=css` 
- Ran all JavaScript tests: `./tools/test-js-with-node`  
- Manual testing: Verified dropdown appears disabled for spectators in channel topics list
- Confirmed no impact on logged-in users
- Verified visual consistency with recent conversations view

**Screenshots **
=> Before:
dark theme : <img width="1608" height="239" alt="Screenshot from 2025-11-06 19-15-59" src="https://github.com/user-attachments/assets/addc541c-002b-4fb5-9440-085f99ee62e8" />
light theme : <img width="1608" height="239" alt="Screenshot from 2025-11-06 19-16-25" src="https://github.com/user-attachments/assets/fe969353-7ea2-457c-a062-a5716901638e" />
=> After:
dark theme : <img width="1758" height="328" alt="Screenshot from 2025-11-06 19-18-56" src="https://github.com/user-attachments/assets/764559ac-6bcc-45a2-a9bc-165cad074281" />
light theme : <img width="1758" height="328" alt="Screenshot from 2025-11-06 19-18-38" src="https://github.com/user-attachments/assets/446d43e3-5ad0-40b0-b027-6506962a9a07" />

<details>
<summary>Self-review checklist</summary>

- [x] Self-reviewed the changes for clarity and maintainability (variable names, code reuse, readability, etc.).
- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.
- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.
- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.

</details>

alya commented:
@amanagr Could you please check if this is the right approach? I haven't tested.

amanagr commented on web/styles/inbox.css:
```diff
@@ -787,3 +767,8 @@
+.dropdown-widget-disabled-for-spectators #inbox-filter_widget {
+    cursor: not-allowed;
+    opacity: 0.5;
+}
```
Let's use `.dropdown-widget-disabled-for-spectators .dropdown-widget-button` and move it near related CSS for `.dropdown-widget-button` in `app_components.css`.

Hell1213 commented:
hey @amanagr Done! I've moved the CSS to app_components.css near the .dropdown-widget-button definition and removed the view-specific versions from both inbox.css and recent_view.css as requested.

amanagr commented:
Thanks! Looks good to me. Tested manually both recent view and inbox channel topic list.

timabbott commented on web/styles/inbox.css:
```diff
@@ -767,3 +767,8 @@
+
+.dropdown-widget-disabled-for-spectators #inbox-filter_widget {
+    cursor: not-allowed;
+    opacity: 0.5;
```
I guess it looks like we were using 0.5 previously in some views. Can you provide screenshots of 0.5 vs 0.7 for this feature?

Hell1213 commented:
> The next step is to address this:
> 
> > Can you provide screenshots of 0.5 vs 0.7 for this feature?

Yeah sure , I'll provide screenshot for both

alya commented:
0.5 looks more clearly disabled to me in these screenshots, maybe because this dropdown is next to a field with a placeholder that itself has reduced opacity. @timabbott what do you think?

timabbott commented:
Yeah I think 0.5 is good. Maybe we want to align towards 0.5 for "disabled", 0.7 for placeholders and the like.
